### PR TITLE
Port #48978 forward to the revised FAR code

### PR DESCRIPTION
**Pull request by amcasey**
The old regression test didn't catch the corresponding bug in the new code because it hit the fast path where there's only a single project.

### CODE REVIEW

**Comment by amcasey** [comment]
Fixes https://github.com/microsoft/tsserverfuzzer/issues/415

File: src/server/session.ts [hunk 1]
```diff
@@ -380,7 +380,7 @@ namespace ts.server {
         // correct results to all other projects.
 
         const defaultProjectResults = perProjectResults.get(defaultProject)!;
-        if (defaultProjectResults[0].references[0].isDefinition === undefined) {
+        if (defaultProjectResults?.[0].references[0]?.isDefinition === undefined) {
```
Comment (amcasey) [inline src/server/session.ts:383]: If there are no results at all, `defaultProjectResults` will be undefined.  If, as for some string literals, there's a definition but no references, `references[0]` will be undefined.  Both are expected and clearing the `isDefinition` flag is (trivially) correct for both.

Comment (sheetalkamat) [reply src/server/session.ts:383]:
```suggestion
        const defaultProjectResults = perProjectResults.get(defaultProject);
        if (defaultProjectResults?.[0].references[0]?.isDefinition === undefined) {        
```

Comment (amcasey) [reply src/server/session.ts:383]: 🤦‍♂️

Comment (amcasey) [reply src/server/session.ts:383]: Applying your fix directly doubled up the declaration.  I've force-pushed a clean commit.

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**